apis/v1alpha1/greptimedbcluster_types.go

```diff
@@ -322,6 +323,93 @@ type GreptimeDBClusterSpec struct {
 	// The global logging configuration for all components. It can be overridden by the logging configuration of individual component.
 	// +optional
 	Logging *LoggingSpec `json:"logging,omitempty"`
+
+	// Monitoring is the specification for monitor bootstrapping. It will create a standalone greptimedb instance to monitor the cluster.
+	// +optional
+	Monitoring *MonitoringSpec `json:"monitoring,omitempty"`
+}
+
+// MonitoringSpec is the specification for monitor bootstrapping. It will create a standalone greptimedb instance to monitor the cluster.
+type MonitoringSpec struct {
+	// Enabled indicates whether to enable the monitoring service.
+	// +required
+	Enabled bool `json:"enabled"`
+
+	// The specification of the standalone greptimedb instance.
+	// +optional
+	Standalone *GreptimeDBStandaloneSpec `json:"standalone,omitempty"`
+
+	// The specification of cluster logs collection.
+	// +optional
+	LogsCollection *LogsCollectionSpec `json:"logsCollection,omitempty"`
+
+	// The specification of the vector instance.
+	// +optional
+	Vector *VectorSpec `json:"vector,omitempty"`
+}
+
+// LogsCollectionSpec is the specification for cluster logs collection.
+type LogsCollectionSpec struct {
+	// The specification of the log pipeline.
+	// +optional
+	Pipeline *LogPipeline `json:"pipeline,omitempty"`
+}
+
+func (in *LogsCollectionSpec) GetPipeline() *LogPipeline {
+	if in != nil {
+		return in.Pipeline
+	}
+	return nil
+}
+
+// LogPipeline is the specification for log pipeline.
+type LogPipeline struct {
+	// The content of the pipeline configuration file in YAML format.
+	// +optional
+	Data string `json:"data,omitempty"`
+}
+
+func (in *LogPipeline) GetData() string {
+	if in != nil {
+		return in.Data
+	}
+	return ""
+}
+
+// VectorSpec is the specification for vector instance.
+type VectorSpec struct {
+	// The image of the vector instance.
+	// +optional
+	Image string `json:"image,omitempty"`
+
+	// The resources of the vector instance.
+	// +optional
+	Resource corev1.ResourceRequirements `json:"resource,omitempty"`
```

Comment:
Can we set the request and limit of vector? It seems that we can only set the request.